src/main/scala/com/cloudera/sparkts/TimeSeries.scala

```diff
@@ -206,6 +206,16 @@ class TimeSeries[K](val index: DateTimeIndex, val data: DenseMatrix[Double],
     }
   }
 
+  def toSamples(): IndexedSeq[(DateTime, Vector[Double])] =
+  {
+    (0 until data.rows).map(rowIndex => (index.dateTimeAtLoc(rowIndex), data(rowIndex, ::).inner.toVector))
+  }
+
+  def toRowSequence(): IndexedSeq[(Int, Vector[Double])] =
```

Comment:
Given that this is a TimeSeries object, and that the data is a DenseMatrix, I think there are at least 2 "transformations" that need to be made to be able to call zipWithIndex on it. In which case it's more elegant to be able to just call toRowSequences() on the TimeSeries object. But maybe I'm missing the more concise way to do it. What did you have in mind?